bitrepository-integrity-service/src/test/java/org/bitrepository/integrityservice/web/PillarDetailsDTOTest.java

```diff
@@ -0,0 +1,24 @@
+package org.bitrepository.integrityservice.web;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+
+import static org.testng.Assert.*;
+
+public class PillarDetailsDTOTest {
+    private final ObjectMapper objectMapper = new ObjectMapper(); // Use ObjectMapper for JSON
+
+    @Test
+    void testSerialization() throws IOException {
+        PillarDetailsDTO dto = new PillarDetailsDTO("id", "pillar", "TYPE", "admin");
+
+        String jsonOutput = objectMapper.writeValueAsString(dto);
+
+        assertTrue(jsonOutput.contains("\"pillarID\":\"id\""));
```

Comment:
A bit strict since JSON syntax has some freedoms, for instance presence of spaces. I suggest we let it stand for now and only do something about it if it turns out to pose a problem.

Reply:
I suppose I am assuming that Jackson would never use those freedoms when serializing, but I don't see why I wouldn't assume that. But yeah, one could do some trimming and stuff before doing the assertions. I'll let it be for now.